types/socket.io/index.d.ts

```diff
@@ -45,6 +45,31 @@ interface SocketIOStatic {
      */
     (opts: SocketIO.ServerOptions): SocketIO.Server;
 
+    /**
+     * Default Server constructor
+     */
+    new (): SocketIO.Server;
+
+    /**
+     * Creates a new Server
+     * @param srv The HTTP server that we're going to bind to
+     * @param opts An optional parameters object
+     */
+    new (srv: any, opts?: SocketIO.ServerOptions): SocketIO.Server;
```

Comment:
Shouldn't it be srv: Server from 'http' ?

Reply:
I agree, but I think it's out of scope for this PR as the `any` type was not introduced with these additions. The type should also be improved to `Server from 'http'` at the following places:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ffc7341b598c33b875c25006047a02960d79bd9d/types/socket.io/index.d.ts#L33

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ffc7341b598c33b875c25006047a02960d79bd9d/types/socket.io/index.d.ts#L58

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ffc7341b598c33b875c25006047a02960d79bd9d/types/socket.io/index.d.ts#L191

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ffc7341b598c33b875c25006047a02960d79bd9d/types/socket.io/index.d.ts#L204

I could add it in this PR if you want me to do that, but I think the type change deserves its own PR.
I'll gladly create that separate PR.

Reply:
I agree. I will appreciate creating separate PR to address that.